pkg/controller/machine.go

```diff
@@ -775,14 +832,18 @@ func (c *controller) updateMachineStatus(
 	currentStatus v1alpha1.CurrentStatus,
 ) (*v1alpha1.Machine, error) {
 	// Get the latest version of the machine so that we can avoid conflicts
-	clone, err := c.controlMachineClient.Machines(machine.Namespace).Get(machine.Name, metav1.GetOptions{})
+	machine, err := c.controlMachineClient.Machines(machine.Namespace).Get(machine.Name, metav1.GetOptions{})
 	if err != nil {
 		return machine, err
 	}
+	clone := machine.DeepCopy()
```

Comment:
```suggestion
	latestMachine, err := c.controlMachineClient.Machines(machine.Namespace).Get(machine.Name, metav1.GetOptions{})
	if err != nil {
		return machine, err
	}
	clone := latestMachine.DeepCopy()
```
We need to have the old value of the `machine` here as there is a call below to retry with `machine` object and if this call fails, the value of machine becomes `nil` and would keep failing.

Reply:
I'd guess, the error should be returned along with that, and should be taken care.
But a good point, to have diff name than argument as we had discussed, as a pointer may mess with the original flow.

Reply:
Yes, i think even you are right eventually the call but would through. But i guess better safe :D